### PR TITLE
docs(new-tutorial): add routing unit

### PR DESCRIPTION
**Pull request by aliemir**
Added the routing unit and mentioned:

- Router integrations
- Defining resources
- Defining routes
- Navigation and Redirection
- authentication
- reading parameters from url
- Sync with location

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: documentation/tutorial/routing/intro/react-router/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+- Refine won't interfere with your router's way of handling navigation. You'll be generating the routes/pages as you would normally do with your router.
+
+- Providing a router provider to Refine will unlock many features without giving up any of your router's features.
+
+:::
+
+This unit will cover the following topics:
+
+- Refine's resource system and how to use it,
```
Comment (omeraplak) [inline documentation/tutorial/routing/intro/react-router/index.md:25]:
```suggestion
- Refine's resource concept and how to use it,
```

**Review by omeraplak — COMMENTED** [review]
We could consider changing the order of 'Inferring' and 'Navigation' in the orders of our sections. This way, users can see and test the 'Inferring' feature to understand how it works. Additionally, we can also write to the page the return values from the 'useParsed()' hook.

File: documentation/tutorial/routing/inferring-parameters/react-router/index.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+title: Inferring Parameters
+---
+
+import { Sandpack, AddInferenceToEditProduct, AddInferenceToCreateProduct, AddInferenceToShowProduct, AddInferenceToListProducts } from "./sandpack.tsx";
+
+<Sandpack>
+
+Now we've learned about the `useNavigation` hook and how to handle navigation with Refine. In this step, we'll be updating components to benefit from the parameter inference of Refine.
+
+Refine when integrated with a router provider, infers the parameters from the route definitions and uses them in its hooks and components. This way, you don't need to pass the `resource`, `id` and `action` parameters to the hooks and components manually.
```
Comment (necatiozmen) [inline documentation/tutorial/routing/inferring-parameters/react-router/index.md:11]:
```suggestion
When integrated with a router provider, Refine infers the parameters from route definitions and incorporates them into its hooks and components, eliminating the need for manual passing of `resource`, `id` and `action` parameters.
```

File: documentation/tutorial/routing/inferring-parameters/react-router/index.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+};
+```
+
+<AddInferenceToListProducts />
+
+## Updating the `ShowProduct` Component
+
+Let's update our `<ShowProduct />` component and omit the `resource` and `id` parameters. Remember that previously we've hard-coded the `id` parameter. Now we'll be letting Refine to infer the `id` parameter from the route definition and dynamically fetch the product.
+
+We'll also switch from `useOne` hook to `useShow` hook. The `useShow` hook is a wrapper around the `useOne` hook and provides inference abilities unlike the `useOne` hook which requires the `resource` and `id` parameters to be passed explicitly.
```
Comment (necatiozmen) [inline documentation/tutorial/routing/inferring-parameters/react-router/index.md:53]:
```suggestion
We'll also start using `useShow` hook which is wrapper around `useOne`. Unlike the useOne hook, it offers inference capabilities, eliminating the need to explicitly pass `resource` and `id` parameters
```

Comment (necatiozmen) [reply documentation/tutorial/routing/inferring-parameters/react-router/index.md:53]: Maybe adding links for useShow?

File: documentation/tutorial/routing/redirects/react-router/index.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+title: Redirects
+---
+
+import { Sandpack, AddRedirectsToAuthProvider, AddCustomRedirectToCreate, AddCustomRedirectToEdit } from "./sandpack.tsx";
+
+<Sandpack>
+
+Now we've updated our components to benefit from the parameter inference of Refine. In this step, we'll be learning about the redirects and how to benefit from them in our forms and auth provider.
+
+Refine can manage redirects automatically for you. After a successful form submission, Refine will try to redirect the user to the appropriate page. This is the list page of the target resource by default. However, you can customize this behavior by providing a `redirect` parameter to the `useForm` hook.
```
Comment (necatiozmen) [inline documentation/tutorial/routing/redirects/react-router/index.md:11]: This is the same paragraph with the section below : "Redirecting After Form Submission". Need to get rid one of them

File: documentation/tutorial/routing/redirects/react-router/index.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+};
+```
+
+<AddCustomRedirectToEdit />
+
+### Continue to Edit the Record After Creation
+
+Let's update our `<CreateProduct />` component and provide a `redirect` parameter to let users continue to edit the created product after a successful form submission.
+
+Try to update your `src/create-product.tsx` file with the following lines:
```
Comment (necatiozmen) [inline documentation/tutorial/routing/redirects/react-router/index.md:54]: Same thing here we can replace this with Let's update our... sentence

**Review by omeraplak — APPROVED** [review]

File: documentation/tutorial/routing/inferring-parameters/react-router/index.md (hunk 3 above)
Comment (aliemir) [inline documentation/tutorial/routing/inferring-parameters/react-router/index.md:53]:
```suggestion
We'll also switch from `useOne` hook to [`useShow`](/docs/data/hooks/use-show) hook. The `useShow` hook is a wrapper around the `useOne` hook and provides inference abilities unlike the `useOne` hook which requires the `resource` and `id` parameters to be passed explicitly.
```

File: documentation/tutorial/routing/navigation/react-router/index.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+You can always use the preferred methods of your routing library to navigate between pages. Refine's navigation hooks are helpers to make it easier to navigate between any action of any resource.
+
+:::
+
+We'll use the `useNavigation` hook and create buttons to navigate to the create, edit and show pages of the products. Additionally we'll provide a link to the list page of the products in the `<Header />` component.
+
+## Adding a Link to the List Page and to the Create Page
+
+We'll be using the `useNavigation` hook from `@refinedev/core` and the `<Link />` component of the `react-router-dom` library to create links to the list page and the create page of the products. Let's update our `<Header />` component and add a link to the list page of the products.
```
Comment (aliemir) [inline documentation/tutorial/routing/navigation/react-router/index.md:21]:
```suggestion
We'll be using the [`useNavigation`](/docs/routing/hooks/use-navigation) hook from `@refinedev/core` and the `<Link />` component of the `react-router-dom` library to create links to the list page and the create page of the products. Let's update our `<Header />` component and add a link to the list page of the products.
```

File: documentation/tutorial/routing/redirects/react-router/index.md [hunk 7]
```diff
@@ -2,10 +2,126 @@
...
 
-import { Sandpack } from "./sandpack.tsx";
+import { Sandpack, AddRedirectsToAuthProvider, AddCustomRedirectToCreate, AddCustomRedirectToEdit } from "./sandpack.tsx";
 
 <Sandpack>
 
-Your content here.
+Now we've updated our components to benefit from the parameter inference of Refine. In this step, we'll be learning about the redirects and how to benefit from them in our forms and auth provider.
+
+Refine can manage redirects automatically for you. After a successful form submission, Refine will try to redirect the user to the appropriate page. This is the list page of the target resource by default. However, you can customize this behavior by providing a `redirect` parameter to the `useForm` hook.
```
Comment (aliemir) [inline documentation/tutorial/routing/redirects/react-router/index.md:11]:
```suggestion
Refine can manage redirects automatically for you. After a successful form submission, Refine will try to redirect the user to the appropriate page.
```

File: documentation/tutorial/routing/resource-definition/react-router/index.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
+<Sandpack>
+
+So far, we've integrated our authentication logic into our routes. In this step, we'll be creating routes for these components and define our resources to inform Refine about their corresponding routes.
+
+To learn more about the Resource concept, refer to the [Resource Concept](/docs/guides-concepts/general-concepts/#resource-concept) section in the General Concepts guide.
+
+## Creating Routes
+
+We've wrapped our routes with the `<Authenticated />` component in the previous step. Now we'll be creating routes under the `/products` path to place our components.
```
Comment (aliemir) [inline documentation/tutorial/routing/resource-definition/react-router/index.md:15]:
```suggestion
We've wrapped our routes with the [`<Authenticated />`](/docs/authentication/components/authenticated) component in the previous step. Now we'll be creating routes under the `/products` path to place our components.
```

File: documentation/tutorial/routing/navigation/react-router/index.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+Now we've set up our routes and resources. In this step, we'll be learning about Refine's navigation helpers and how to use them in our app.
+
+:::tip
+
+You can always use the preferred methods of your routing library to navigate between pages. Refine's navigation hooks are helpers to make it easier to navigate between any action of any resource.
+
+:::
+
+We'll use the `useNavigation` hook and create buttons to navigate to the create, edit and show pages of the products. Additionally we'll provide a link to the list page of the products in the `<Header />` component.
```
Comment (aliemir) [inline documentation/tutorial/routing/navigation/react-router/index.md:17]:
```suggestion
We'll use the [`useNavigation`](/docs/routing/hooks/use-navigation) hook and create buttons to navigate to the create, edit and show pages of the products. Additionally we'll provide a link to the list page of the products in the `<Header />` component.
```

File: documentation/tutorial/routing/inferring-parameters/react-router/index.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+};
+```
+
+<AddInferenceToListProducts />
+
+## Updating the `ShowProduct` Component
+
+Let's update our `<ShowProduct />` component and omit the `resource` and `id` parameters. Remember that previously we've hard-coded the `id` parameter. Now we'll be letting Refine to infer the `id` parameter from the route definition and dynamically fetch the product.
+
+We'll also start using `useShow` hook which is wrapper around `useOne`. Unlike the useOne hook, it offers inference capabilities, eliminating the need to explicitly pass `resource` and `id` parameters
```
Comment (aliemir) [inline documentation/tutorial/routing/inferring-parameters/react-router/index.md:53]:
```suggestion
We'll also start using [`useShow`](/docs/data/hooks/use-show) hook which is wrapper around `useOne`. Unlike the useOne hook, it offers inference capabilities, eliminating the need to explicitly pass `resource` and `id` parameters
```

File: documentation/tutorial/routing/inferring-parameters/react-router/index.md (hunk 3 above)
Comment (aliemir) [inline documentation/tutorial/routing/inferring-parameters/react-router/index.md]:
```suggestion
We'll also switch from `useOne` hook to [`useShow`](/docs/data/hooks/use-show) hook. The `useShow` hook is a wrapper around the `useOne` hook and provides inference abilities unlike the `useOne` hook which requires the `resource` and `id` parameters to be passed explicitly.
```

**Pull request merged**